pandas/core/window/rolling.py

```diff
@@ -2102,12 +2103,101 @@ class RollingGroupby(WindowGroupByMixin, Rolling):
     Provide a rolling groupby implementation.
     """
 
+    def _apply(
+        self,
+        func: Callable,
+        center: bool,
+        require_min_periods: int = 0,
+        floor: int = 1,
+        is_weighted: bool = False,
+        name: Optional[str] = None,
+        use_numba_cache: bool = False,
+        **kwargs,
+    ):
+        result = Rolling._apply(
+            self,
+            func,
+            center,
+            require_min_periods,
+            floor,
+            is_weighted,
+            name,
+            use_numba_cache,
+            **kwargs,
+        )
+        # _wrap_outputs does not know about what the result index should be
+        grouped_object_index = self._groupby._selected_obj.index
+        grouped_index_name = [grouped_object_index.name]
+        groupby_keys = [grouping.name for grouping in self._groupby.grouper._groupings]
+        result_index_names = groupby_keys + grouped_index_name
+
+        result_index_data = []
+        for key, values in self._groupby.grouper.indices.items():
+            for value in values:
+                if not is_list_like(key):
+                    data = [key, grouped_object_index[value]]
+                else:
+                    data = [*key, grouped_object_index[value]]
+                result_index_data.append(tuple(data))
+
+        result_index = MultiIndex.from_tuples(
+            result_index_data, names=result_index_names
+        )
+        result.index = result_index
+        return result
+
     @property
     def _constructor(self):
         return Rolling
 
-    def _gotitem(self, key, ndim, subset=None):
+    def _create_blocks(self):
+        """
+        Split data into blocks & return conformed data.
+        """
+        # Ensure the object we're rolling over is monotonically sorted relative
+        # to the groups
+        obj = self._selected_obj.take(
+            np.concatenate(list(self._groupby.grouper.indices.values()))
+        )
+
+        # filter out the on from the object
```

Comment:
*maybe* better to call super()._create_blocks(obj) (e.g. add obj as an optional arg that defaults to self._selected_obj)